include/openPMD/RecordComponent.hpp

```diff
@@ -291,6 +293,36 @@ class RecordComponent : public BaseRecordComponent
     template <typename T>
     void storeChunk(std::shared_ptr<T[]> data, Offset offset, Extent extent);
 
+    /** Store a chunk of data from a chunk of memory, unique pointer version.
+     *
+     * @param data   Preallocated, contiguous buffer, large enough to read the
+     *               the specified data from it.
+     *               The unique pointer must own and manage the buffer.
+     *               Optimizations might be implemented based on this
+     *               assumption (e.g. further deferring the operation if the
+     *               backend is the unique owner).
+     *               For raw pointers, use storeChunkRaw().
+     * @param offset Offset within the dataset.
+     * @param extent Extent within the dataset, counted from the offset.
+     */
+    template <typename T>
+    void storeChunk(OpenpmdUniquePtr<T> data, Offset offset, Extent extent);
```

Comment:
We could theoretically delete this overload, as `OpenpmdUniquePtr`s are also accepted by the `std::unique_ptr` overload. However, `OpenpmdUniquePtr` is useful for specifying custom destructors, and having its own overload will promote its usage.
Apart from this, `OpenpmdUniquePtr` is needed internally.